tests/unit-tests/gconstruct/test_construct_graph.py

```diff
@@ -1705,6 +1705,46 @@ def test_gc():
     assert not os.path.isdir("/tmp_featurewrapper2"), \
         "Directory /tmp_featurewrapper2 should not exist after gc"
 
+
+def test_homogeneous():
```

Comment:
Please also add a test for is_homogeneous() function.

Reply:
Add the test here in the test_homogeneous() as we need to reuse the config here